src/pages/App.tsx

```diff
@@ -165,8 +169,21 @@ export default function App() {
   }, [isDarkMode])
 
   useEffect(() => {
+    // If we're not in the transition period to UniswapX opt-out, set the router preference to whatever is specified.
+    if (!isUniswapXDefaultEnabled) {
+      user.set(CustomUserProperties.ROUTER_PREFERENCE, routerPreference)
+      return
+    }
+
+    // In the transition period, override the stored API preference to UniswapX if the user hasn't opted out.
+    if (routerPreference === RouterPreference.API && !userOptedOutOfUniswapX) {
+      user.set(CustomUserProperties.ROUTER_PREFERENCE, RouterPreference.X)
```

Comment:
oh is this because we will eventually pass through UniswapX as the preference? would it make more sense to just override their actual router preference in this case

Reply:
yeah, we eventually need to override the stored preference. i'm doing it this way because i wanted to write a single redux migration to migrate everyone's preferences at once, but since we're doing a partial rollout for a couple weeks i need to wait until it's at 100% to enact the migration

Reply:
smart!